src/figuresClasses.ts

```diff
@@ -1,19 +1,86 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+
+  ) {
+    if (!this.isCorrect()) {
+      throw new Error('Side length is not correct');
+    }
+  }
+
+  isCorrect(): boolean {
+    const { a, b, c } = this;
+
+    if ((a <= 0 || b <= 0 || c <= 0)
+      || (a + b <= c
+        || a + c <= b || b + c <= a)) {
+      return false;
```

Comment:
Keep the right indentation and use short syntax
```suggestion
    return (
      (a <= 0 || b <= 0 || c <= 0) 
      || (a + b <= c || a + c <= b || b + c <= a)
    )
```